### PR TITLE
added support for using TLS with MQTT

### PR DESCRIPTION
**Pull request by LordGuilly**
3 new entries in the config section, for setting file paths for
        - Root CA
        - Client Certificate
        - Client Private Key

If these are defined, the MQTT connection uses TLS (needed for most Cloud providers)

### CODE REVIEW

File: sd-card/config/config.ini [hunk 1]
```diff
@@ -62,6 +62,9 @@ CheckDigitIncreaseConsistency = false
 RetainMessages = false
 HomeassistantDiscovery = false
 ;MeterType = other
+;CACert = /config/certs/filepath
```
Comment (caco3) [inline sd-card/config/config.ini:65]: I would add default file names instead of just `filepath`.

File: sd-card/html/edit_config_param.html [hunk 2]
```diff
@@ -753,6 +753,39 @@ <h4>
 			<td>$TOOLTIP_MQTT_password</td>
 		</tr>
 
+		<tr class="MQTTItem">
```
Comment (caco3) [inline sd-card/html/edit_config_param.html:756]: The new parameters should be marked as expert parameters, see https://github.com/jomjol/AI-on-the-edge-device/pull/2651/files#diff-a5697edb14c8de88b221309f55facd517353ada57d0e18b22b2ca04306d79317R564

File: sd-card/config/config.ini (hunk 1 above)
Comment (LordGuilly) [reply sd-card/config/config.ini]: updated with more specfici names (yet still a bit generic)

File: sd-card/html/edit_config_param.html (hunk 2 above)
Comment (LordGuilly) [reply sd-card/html/edit_config_param.html]: marked the 3 as expert mode

File: code/components/jomjol_mqtt/interface_mqtt.cpp [hunk 3]
```diff
@@ -169,6 +173,10 @@ static esp_err_t mqtt_event_handler_cb(esp_mqtt_event_handle_t event) {
             else if (event->error_handle->connect_return_code == MQTT_CONNECTION_REFUSE_NOT_AUTHORIZED) {
                 LogFile.WriteToFile(ESP_LOG_ERROR, TAG, "Connection refused, not authorized. Check username/password (0x05)");
             }
+            else {
+                LogFile.WriteToFile(ESP_LOG_ERROR, TAG, "UNHANDLED ERROR");
```
[the remark this answers is not in the document]
Comment (LordGuilly) [reply code/components/jomjol_mqtt/interface_mqtt.cpp]: fixed, not it replicates the log error

**Comment by LordGuilly** [comment]
updated with the comments, will update the docs in the next spare time slot :-)

**Comment by caco3** [comment]
I added the parameter doc pages, please have a look and let me know if the description needs improvement/changes.
https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#section-mqtt

It looks ok for me, but I can not test it myself.

**Comment by LordGuilly** [comment]
I created the PR for the docs, sorry it took me so long - I just added some comments about how each file is used. Usually the 3 are needed, although some configurations might need some :-)

**Pull request merged**